### ISSUE TITLE
Is there any significance performance difference by using these scripts instead of the perl scripts

### ISSUE THREAD

**Issue by vanga** [osiegmar/cloudwatch-mon-scripts-python#4]
This issue is just a query.
First of all, thanks for these scripts.

These scripts seem very light weight compared to the perl scripts (atleast setting them up). 

Is there any performance difference bw the perl and these scripts? May I know what was your motivation for writing these?

I often thought that setting up perl scripts was heavy, too many packages to install, CPU used to always hit 100% on EC2 micro instances.

Thanks

**Comment by osiegmar** [comment]
My main motivation to write these scripts was, that the original perl version of AWS CloudWatchMonitoringScripts I used (version 1.1.0) were unable to handle the version 4 authentication scheme (http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) of new AWS regions (such as Frankfurt). This has changed with their version 1.2 released on 2015-02-25.

I offered my version to AWS, but they decided to update their Perl version instead.

Without doing a scientific analysis, it seem that my version is about 30% faster and has only very little dependencies (boto). I'm using my version on several t2.micro instances and a single call to `mon-put-instance-stats.py --mem-util --disk-space-util --disk-path /` takes about 140 ms. I couldn't measure the CPU load for typical execution interval (once every minute).

I'm not using the Perl version anymore, because of its heavy dependency footprint.
